eslint.config.js

```diff
@@ -0,0 +1,169 @@
+      'src/**/*.js',
+
+      '.github/styles/',
+      '**/__fixtures__/',
+    ],
+  },
+  {
+    files: ['**/*.?(c|m)ts?(x)'],
+    rules: {
+      // `interface` and `type` have different use cases, allow both
+      '@typescript-eslint/consistent-type-definitions': 'off',
+
+      // Ignore underscore-prefixed unused variables (mirrors tsc behavior)
+      '@typescript-eslint/no-unused-vars': [
+        'error',
+        {
+          args: 'all',
+          argsIgnorePattern: '^_',
+          caughtErrors: 'all',
+          caughtErrorsIgnorePattern: '^_',
+          destructuredArrayIgnorePattern: '^_',
+          ignoreRestSiblings: true,
+          varsIgnorePattern: '^_',
+        },
+      ],
+    },
+  },
+  {
+    rules: {
+      'no-restricted-imports': [
+        'error',
+        {
+          paths: [
+            {
+              name: 'node:process',
+              importNames: ['cwd'],
+              message: 'Use `command.workingDir` instead.',
+            },
+            {
+              name: 'process',
+              importNames: ['cwd'],
+              message: 'Use `command.workingDir` instead.',
+            },
+
+            {
+              name: 'chalk',
+              message:
+                'Use the safe chalk import that handles colors for json output: `import { chalk } from "src/utils/command-helpers.js"`',
+            },
+          ],
+        },
+      ],
+      'no-restricted-properties': [
+        'error',
+        {
+          object: 'process',
+          property: 'cwd',
+          message: '`process.cwd` is not permitted; use `command.workingDir` instead.',
+        },
+      ],
+    },
+  },
+
+  // Tests
+  {
+    files: ['**/*.test.?(c|m)[jt]s?(x)'],
+    plugins: { vitest },
+    rules: {
+      ...vitest.configs.recommended.rules,
+
+      'vitest/expect-expect': [
+        'error',
+        {
+          assertFunctionNames: [
+            // Defaults
+            'assert',
+            'expect',
+
+            // Fix issue where text-context-specific `expect()` calls trigger false positive
+            't.expect',
+            'ctx.expect',
+            'context.expect',
+
+            // Custom assertion functions
+            'assertNetlifyToml',
+          ],
+        },
+      ],
+    },
+  },
+
+  // XXX: Temporarily disabled rules: These rules are disabled because we have offending code that we haven't yet fixed.
+  {
+    rules: {
+      // Empty functions and blocks are useful (e.g `noop() {}`, `catch {}`) but can mask unintentionally omitted
+      // implementation. We should add explanatory comments like `// intentionally empty` and `// ignore error` in these
+      // scenarios to communicate intent.
+      'no-empty': 'off',
+      '@typescript-eslint/no-empty-function': 'off',
+
+      'n/no-unsupported-features/node-builtins': [
+        'error',
+        {
+          ignores: ['FormData', 'ReadableStream', 'Response', 'fetch', 'fs/promises.cp'],
```

Comment:
Can we leave a comment explaining why these are ignored? I personally have no idea

Reply:
I can add a comment if you think it would be helpful, but I think it's pretty self-evident if you consult the rule definition: These are unstable features, but we use them.